### DISCUSSION TITLE
Table performance optimization

### DISCUSSION THREAD

**Discussion by AsuraKev** [ant-design/ant-design#44120] [General]
Hello,

Currently we experienced significant Table performance issue when pagination is turned off and showing more than 100 row. The lag is real when the rowSelection is enabled and it affects the UI responsiveness as well.

**shouldCellUpdate** column config is not really doing anything even when we hardcode to return false.

I believe this is the library fundamental issue and hope we can address this issue over the next  upcoming releases.

Thanks a lot Antd team, very good library :)

**Comment by olegkobzar** [comment]
Are there any changes? Now, when the checkbox state changes, the entire table is re-render

**Reply by AsuraKev** [reply]
Don’t think so. Have been using virtual table for now.

**Reply by olegkobzar** [reply]
What library do you use?

**Reply by AsuraKev** [reply]
Still Antd table with props virtual

**Comment by AsuraKev** [comment]
Could this issue potentially be resolved by React 19 compiler?

**Comment by SmashPlusUltra** [comment]
It's strange to have this problem. It's almost common sense to try to reduce re-renders or at least give you the opportunity to turn it off.
https://ant.design/docs/blog/render-times
